dask/dataframe/_pyarrow.py

```diff
@@ -52,52 +61,66 @@ def is_object_string_dataframe(x):
     )
 
 
-def to_pyarrow_string(df):
+def _to_string_dtype(df, dtype_check, index_check, string_dtype):
     if not (is_dataframe_like(df) or is_series_like(df) or is_index_like(df)):
         return df
 
+    # Guards against importing `pyarrow` at the module level (where it may not be installed)
+    if string_dtype == "pyarrow":
+        string_dtype = pd.StringDtype("pyarrow")
+
     # Possibly convert DataFrame/Series/Index to `string[pyarrow]`
     dtypes = None
     if is_dataframe_like(df):
         dtypes = {
-            col: pd.StringDtype("pyarrow")
-            for col, dtype in df.dtypes.items()
-            if is_object_string_dtype(dtype)
+            col: string_dtype for col, dtype in df.dtypes.items() if dtype_check(dtype)
         }
-    elif is_object_string_dtype(df.dtype):
-        dtypes = pd.StringDtype("pyarrow")
+    elif dtype_check(df.dtype):
+        dtypes = string_dtype
 
     if dtypes:
         df = df.astype(dtypes, copy=False)
 
     # Convert DataFrame/Series index too
-    if (is_dataframe_like(df) or is_series_like(df)) and is_object_string_index(
-        df.index
-    ):
+    if (is_dataframe_like(df) or is_series_like(df)) and index_check(df.index):
         if isinstance(df.index, pd.MultiIndex):
             levels = {
-                i: level.astype(pd.StringDtype("pyarrow"))
+                i: level.astype(string_dtype)
                 for i, level in enumerate(df.index.levels)
-                if is_object_string_dtype(level.dtype)
+                if dtype_check(level.dtype)
             }
             # set verify_integrity=False to preserve index codes
             df.index = df.index.set_levels(
                 levels.values(), level=levels.keys(), verify_integrity=False
             )
         else:
-            df.index = df.index.astype(pd.StringDtype("pyarrow"))
+            df.index = df.index.astype(string_dtype)
     return df
 
 
+to_pyarrow_string = partial(
+    _to_string_dtype,
+    dtype_check=is_object_string_dtype,
+    index_check=is_object_string_index,
+    string_dtype="pyarrow",
+)
+to_object_string = partial(
+    _to_string_dtype,
+    dtype_check=is_pyarrow_string_dtype,
+    index_check=is_pyarrow_string_index,
+    string_dtype=object,
+)
+
+
 def check_pyarrow_string_supported():
     """Make sure we have all the required versions"""
-    if pa is None:
+    if not PANDAS_GT_200:
         raise RuntimeError(
             "Using dask's `dataframe.convert-string` configuration "
-            "option requires `pyarrow` to be installed."
+            "option requires `pandas>=2.0` to be installed."
         )
-    if not PANDAS_GT_200:
+    if pa is None or Version(pa.__version__) < Version("12.0.0"):
```

Comment:
Note: I'm bumping the minimum version of `pyarrow` required for automatic pyarrow strings to `pyarrow>=12` because otherwise string-backed categoricals won't work with Parquet (xref https://github.com/apache/arrow/issues/33727)